src/sisl/geom/_neighbors/_finder.py

```diff
@@ -3,17 +3,213 @@
 # file, You can obtain one at https://mozilla.org/MPL/2.0/.
 from __future__ import annotations
 
-from typing import Optional, Sequence, Tuple, Union
+from functools import cached_property
+from typing import Literal, Optional, Sequence, Tuple, Union
 
 import numpy as np
 
-from sisl import Geometry
+from sisl import Geometry, Lattice
 from sisl.typing import AtomsIndex
 from sisl.utils import size_to_elements
 
 from . import _operations
 
 
+class Interactions:
+
+    def __init__(self, geometry: Geometry, finder_results, split_indices=None):
+        self.geometry = geometry
+        self.finder_results = finder_results
+        self._split_indices = split_indices
+
+    @property
+    def i(self):
+        return self.finder_results[:, 0]
+
+    @property
+    def j(self):
+        return self.finder_results[:, 1]
+
+    @property
+    def isc(self):
+        return self.finder_results[:, 2:]
+
+    @cached_property
+    def nsc(self):
+        return np.max(np.abs(self.isc), axis=0) * 2 + 1
+
+    @property
+    def j_sc(self):
+        lattice = Lattice(cell=[[1, 0, 0], [0, 1, 0], [0, 0, 1]], nsc=self.nsc)
```

Comment:
Hmm my idea with the raw finder is that all you need to know are the boundary conditions and the finder is the one that helps you determine `nsc`.

My idea was that users would do:

```python
finder = NeighborFinder(geometry)

neighs = finder.find_neighbors()

geometry.set_nsc(neighs.nsc)
```

But perhaps the geometry stored in the `Interactions` class should be a copy of the provided one with the right `nsc`? So that they would take it like:

```python
finder = NeighborFinder(geometry)

neighs = finder.find_neighbors()

geom_with_aux_cell = neighs.geometry
```

Reply:
I can understand the niche caches, but if the finder is used for setting up matrices, then this is not clear. I don't know if the classes should simply fail if the nsc is not compatible...?

Reply:
> I can understand the niche caches, but if the finder is used for setting up matrices, then this is not clear.

What is not clear?

Reply:
that `j_sc` cannot be used

Reply:
But how should it work? `nsc` is determined after finding the neighbors up to a certain R. So it is when you run the finder that you get `nsc`, not before.

So my view is that you should retreive `nsc` from the finder and then initialize a matrix with that. 

Maybe the `Neighbors` class should have a method like:

```python

def set_nsc(self, nsc=None):
    if nsc is None:
        # Compute nsc from the max of isc
    else:
        # Set nsc to whatever the user has passed (maybe have another method
        # that removes further interactions)
```

Reply:
I don't know how it should work, but the ambiguity of doing something like:
```python
geom = ...
H = Hamiltonian(geom)
neighbors = NeighborFinder(geom)
atomneighbor = neighbors[10].j_sc
H[geom.a2o(10), geom.a2o(atomneighbor.j_sc)] = ...
```
is not totally clear. And there is no way to catch it (except when `j_sc` is out of bounds!).
Either, the neighbor finder should respect the geometry's `nsc` and raise an error if something inconsistent is found.
Or we shouldn't allow the `j_sc` (even though I like it).

It also seems weird to create a new geometry based on the neighbors before creating stuff. (minor point).
It is one of those cases where it makes sense for us, but not for end-users.

Reply:
What I would say in that case is that you shouldn't create the matrix until you know what the connectivity is.

I don't understand how expecting users to know what `nsc` should be makes it more logic/usable. In my opinion it should be possible to use the results of the neighbor finder to determine `nsc`. It is basically what `Geometry.optimize_nsc` does right now. If something it would make it more transparent for begginers to understand what `nsc` is.

Reply:
I think we should start by removing `j_sc`. Then we can always figure out how to do things. By doing this we force users to use the `i`, `j` and `isc` properties.

Reply:
Ok, makes sense